hazelcast/src/main/java/com/hazelcast/query/impl/UnorderedIndexStore.java

```diff
@@ -327,12 +327,8 @@ public Object invoke(Comparable value, QueryableEntry entry) {
             if (value == NULL) {
                 return recordsWithNullValue.put(entry.getKeyData(), entry);
             } else {
-                Map<Data, QueryableEntry> records = recordMap.get(value);
-                if (records == null) {
-                    records = new ConcurrentHashMap<>(1, LOAD_FACTOR, 1);
-                    recordMap.put(value, records);
-                }
-                return records.put(entry.getKeyData(), entry);
+                return recordMap.computeIfAbsent(value, x -> new ConcurrentHashMap<>(1, LOAD_FACTOR, 1)).put(entry.getKeyData(),
```

Comment:
There is a difference here between the original code. In the original code ConcurrentHashMap.put is always called but the in PR ConcurrentHashMap.put is called only if it is absent

Reply:
> There is a difference here between the original code. In the original code ConcurrentHashMap.put is always called but the in PR ConcurrentHashMap.put is called only if it is absent

Sorry, I thought I replied to this - I don't think that's the case.

It could've been written as:
```
var recordMap.computeIfAbsent(value, x -> new ConcurrentHashMap<>(1, LOAD_FACTOR, 1));
var.put(blah);
```

It's doing the `put` on the value in the map every time - and the value in the map is retrieved or created if required.